plugins/eos/gs-plugin-eos.c

```diff
@@ -2205,13 +2224,16 @@ add_updates (GsPlugin *plugin,
 					   "com.endlessm.OperatingSystemApp.desktop",
 					   NULL};
 
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	gboolean is_hack_product = g_strcmp0 (priv->product_name, "hack") == 0;
+
 	process_proxy_updates (plugin, list,
 			       framework_proxy_app,
 			       FALSE,
 			       framework_proxied_apps);
 	process_proxy_updates (plugin, list,
 			       hack_proxy_app,
-			       FALSE,
+			       is_hack_product,
```

Comment:
Do we really want the behaviour of gnome-software to differ based on the OS? That doubles the testing space. Could this just be based on whether the Hack proxy app is installed in the first place?

Reply:
Proxy apps aren't really "installed", they're created and shown as updatable just for masking the updates. So their existence is ephemeral.

It's also not trivial to understand whether an app is installed or not in this place (hence the amount of patches just for making sure that we don't show the Hack proxy app if all its apps are already installed and up-to-date).

Reply:
What would happen if `TRUE` was unconditionally passed to this function on both versions of the OS, rather than passing `is_hack_product`?

Reply:
@pwithnall all apps belonging to either proxy apps would end up installed.